The traditional protein research involves strenuous experimental procedures and preforms on individual protein at one time. Consequently, early studies on proteases are commonly limited to small number of underline substrates \cite{:1992aa}. The appearance of recombinant library has expanded our understanding to exo-hydrolytic events greatly by using large recombinant terminal libraries \cite{Gupta:2010aa}. Nevertheless, recombinant library technique is useful \textit{in vitro} settings, reveals little \textit{in vivo} relevance. \textit{In vivo} hydrolytic events have more stringent restrictions, including subcellular localization of substrates and proteases, subcelluar concentrations, presence of various chaperones etc. Given the rise of modern techniques, particular the mass spectrometry has opened up the possibility of validation of \textit{in vivo} hydrolysis events in a high throughput fashion. In addition, with rapidly growing data volume, "big data" \textit{in silico} analysis has started to become popular, in contrary to traditional proteomics research, "big data" analysis focus on extracting features rather than depict the underline mechanisms. 

Revealing the entire collection of substrates of a given protease is important to understand biological relevance of protease in the context of entire organism biology, nevertheless, this is particular difficult in the stream of experimental biochemistry. In this study, we will demonstrate the use of bioinformatics pipeline to mine the substrate degradome of dipeptidyl dipeptidase-4 (DPP4) in human and mice using "big data" approach. 

DPP4 is also known as CD26 cell receptor or adenosine deaminase binding protein (ADA) and is a ubiquitously expressed type II integral membrane protein \cite{Mentlein_1999}. At molecular level, DPP4 has a single 22 amino acid N-terminal hydrophobic domain that serve both signaling purpose and anchorage footing. \cite{Mentlein_1999, Abbott_2002} DPP4 exists both in membrane anchored and free soluble forms, in the soluble form, the trans-membrane domain 